// angular-custom-tour/src/intro-tour/tour.component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var hint_service_1 = require("../hint.service");
var variables_1 = require("../variables");
var TourComponent = /** @class */ (function () {
    function TourComponent(hintService) {
        this.hintService = hintService;
    }
    TourComponent.prototype.ngOnInit = function () {
        this.hintService.register(this.selector, this);
    };
    TourComponent.prototype.showStep = function () {
        this.hintService.showingStep$.next(this);
        this.position = this.position || this.hintService.hintOptions.defaultPosition;
        this.order = +this.order || this.hintService.hintOptions.defaultOrder;
        var highlightedElement = document.getElementById(this.selector);
        if (highlightedElement) {
            highlightedElement.style.zIndex = variables_1.HintConfig.Z_INDEX;
            if (this.hintService.hintOptions.elementsDisabled) {
                this.disableClick(highlightedElement);
            }
            if (this.hintService.hintOptions.applyRelative) {
                highlightedElement.classList.add('hint-relative');
            }
            switch (this.position) {
                case 'top':
                    this.transformClass = 'transformX_50 transformY_100';
                    this.topPos = highlightedElement.offsetTop - this.hintService.hintOptions.defaultLayer;
                    this.leftPos = highlightedElement.offsetLeft + highlightedElement.offsetWidth / 2;
                    break;
                case 'bottom':
                    this.transformClass = 'transformX_50';
                    this.topPos = highlightedElement.offsetTop + highlightedElement.offsetHeight + this.hintService.hintOptions.defaultLayer;
                    this.leftPos = highlightedElement.offsetLeft + highlightedElement.offsetWidth / 2;
                    break;
                case 'left':
                    this.topPos = highlightedElement.offsetTop + highlightedElement.offsetHeight / 2;
                    this.leftPos = highlightedElement.offsetLeft - this.hintService.hintOptions.defaultLayer;
                    this.transformClass = 'transformY_50 transformX_100';
                    break;
                case 'right':
                    this.topPos = highlightedElement.offsetTop + highlightedElement.offsetHeight / 2;
                    this.leftPos = highlightedElement.offsetLeft + highlightedElement.offsetWidth + this.hintService.hintOptions.defaultLayer;
                    this.transformClass = 'transformY_50';
                    break;
                default:
                    throw 'Invalid hint position ->' + this.position;
            }
        }
        else {
            this.topPos = 0;
            this.leftPos = 0;
        }
        this.showme = true;
        this.hasNext = this.hintService.hasNext();
        this.hasPrev = this.hintService.hasPrev();
    };
    TourComponent.prototype.hideStep = function () {
        var highlightedElement = document.getElementById(this.selector);
        if (highlightedElement) {
            highlightedElement.style.zIndex = '0';
            this.enableClick(highlightedElement);
            highlightedElement.classList.remove('hint-relative');
        }
        this.showme = false;
    };
    TourComponent.prototype.exit = function () {
        this.hintService.end();
    };
    TourComponent.prototype.next = function () {
        this.hideStep();
        this.hintService.showNext();
    };
    TourComponent.prototype.prev = function () {
        this.hideStep();
        this.hintService.showPrev();
    };
    TourComponent.prototype.disableClick = function (element) {
        element.classList.add('hint-disabled');
    };
    TourComponent.prototype.enableClick = function (element) {
        element.classList.remove('hint-disabled');
    };
    TourComponent.decorators = [
        { type: core_1.Component, args: [{
                    selector: variables_1.HintConfig.HINT_TAG,
                    template: "<div class=\"intro-tour-hint-wrapper {{transformClass}} step{{order}} {{position}}\"\n  *ngIf=\"showme\" [ngStyle]=\"{'top': topPos+'px', 'left': leftPos+'px'}\" >\n    <div class=\"header\" *ngIf=\"title\">{{title}}</div>\n    <div class=\"content\"><ng-content></ng-content></div>\n    <div class=\"footer\">\n      <a class=\"navigate-btn\" *ngIf=\"hasPrev\" (click)=\"prev()\">&#8592;</a>\n      <a class=\"navigate-btn\" *ngIf=\"hasNext\" (click)=\"next()\">&#8594;</a>\n      <a class=\"navigate-btn\" (click)=\"exit()\">&#10006;</a>\n    </div>\n  </div>",
                },] },
    ];
    /** @nocollapse */
    TourComponent.ctorParameters = function () { return [
        { type: hint_service_1.HintService, },
    ]; };
    TourComponent.propDecorators = {
        'title': [{ type: core_1.Input },],
        'selector': [{ type: core_1.Input },],
        'order': [{ type: core_1.Input },],
        'position': [{ type: core_1.Input },],
    };
    return TourComponent;
}());
exports.TourComponent = TourComponent;
