// angular-custom-tour/src/intro-tour/tour.component.ngfactory.js
"use strict";
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
Object.defineProperty(exports, "__esModule", { value: true });
var i0 = require("@angular/core");
var i1 = require("@angular/common");
var i2 = require("./tour.component");
var i3 = require("../hint.service");
var styles_TourComponent = [];
exports.RenderType_TourComponent = i0.ɵcrt({ encapsulation: 2,
    styles: styles_TourComponent, data: {} });
function View_TourComponent_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'div', [['class',
                'header']], null, null, null, null, null)),
        (_l()(), i0.ɵted(1, null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.title;
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_TourComponent_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'a', [['class',
                'navigate-btn']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.prev() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['←']))], null, null);
}
function View_TourComponent_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'a', [['class',
                'navigate-btn']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.next() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['→']))], null, null);
}
function View_TourComponent_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 21, 'div', [], [[8, 'className', 0]], null, null, null, null)),
        i0.ɵdid(1, 278528, null, 0, i1.NgStyle, [i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngStyle: [0, 'ngStyle'] }, null), i0.ɵpod(2, { 'top': 0, 'left': 1 }),
        (_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TourComponent_2)), i0.ɵdid(5, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵeld(7, 0, null, null, 1, 'div', [['class', 'content']], null, null, null, null, null)), i0.ɵncd(null, 0), (_l()(),
            i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵeld(10, 0, null, null, 10, 'div', [['class', 'footer']], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TourComponent_3)), i0.ɵdid(13, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TourComponent_4)), i0.ɵdid(16, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵeld(18, 0, null, null, 1, 'a', [['class',
                'navigate-btn']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.exit() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['✖'])), (_l()(),
            i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵted(-1, null, ['\n  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = _ck(_v, 2, 0, (_co.topPos + 'px'), (_co.leftPos + 'px'));
        _ck(_v, 1, 0, currVal_1);
        var currVal_2 = _co.title;
        _ck(_v, 5, 0, currVal_2);
        var currVal_3 = _co.hasPrev;
        _ck(_v, 13, 0, currVal_3);
        var currVal_4 = _co.hasNext;
        _ck(_v, 16, 0, currVal_4);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = i0.ɵinlineInterpolate(3, 'intro-tour-hint-wrapper ', _co.transformClass, ' step', _co.order, ' ', _co.position, '');
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_TourComponent_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(16777216, null, null, 1, null, View_TourComponent_1)), i0.ɵdid(1, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef,
            i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null)], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.showme;
        _ck(_v, 1, 0, currVal_0);
    }, null);
}
exports.View_TourComponent_0 = View_TourComponent_0;
function View_TourComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'tour-step', [], null, null, null, View_TourComponent_0, exports.RenderType_TourComponent)), i0.ɵdid(1, 114688, null, 0, i2.TourComponent, [i3.HintService], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
exports.View_TourComponent_Host_0 = View_TourComponent_Host_0;
exports.TourComponentNgFactory = i0.ɵccf('tour-step', i2.TourComponent, View_TourComponent_Host_0, { title: 'title', selector: 'selector',
    order: 'order', position: 'position' }, {}, ['*']);
