// angular-custom-tour/src/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var hint_service_1 = require("./hint.service");
exports.HintService = hint_service_1.HintService;
var common_1 = require("@angular/common");
var core_1 = require("@angular/core");
var hint_component_1 = require("./hint.component");
var tour_component_1 = require("./intro-tour/tour.component");
exports.TourComponent = tour_component_1.TourComponent;
var HintModule = /** @class */ (function () {
    function HintModule() {
    }
    HintModule.forRoot = function () {
        return {
            ngModule: HintModule,
            providers: [
                hint_service_1.HintService,
            ],
        };
    };
    HintModule.decorators = [
        { type: core_1.NgModule, args: [{
                    declarations: [hint_component_1.HintComponent, tour_component_1.TourComponent],
                    exports: [hint_component_1.HintComponent, tour_component_1.TourComponent],
                    imports: [common_1.CommonModule],
                },] },
    ];
    /** @nocollapse */
    HintModule.ctorParameters = function () { return []; };
    return HintModule;
}());
exports.HintModule = HintModule;
